x-pack/plugins/observability_solution/slo/public/hooks/use_fetch_rules_for_slo.ts

```diff
@@ -40,17 +41,20 @@ export function useFetchRulesForSlo({ sloIds = [] }: Params): UseFetchRulesForSl
       try {
         const body = JSON.stringify({
           filter: sloIds.map((sloId) => `alert.attributes.params.sloId:${sloId}`).join(' or '),
-          fields: ['params', 'name'],
           per_page: 1000,
         });
 
         const response = await http.post<RuleApiResponse>(`/internal/alerting/rules/_find`, {
           body,
         });
 
+        const rules = response.data.map((rule) => transformRule(rule)) as Array<
```

Comment:
@shahzad31 I am trying to understand why you added this part here. Was it not working before?

I looked a bit more and found this `rewriteRulesResponseRes` [helper](https://github.com/elastic/kibana/blob/main/x-pack/plugins/triggers_actions_ui/public/application/lib/rule_api/rules_helpers.ts#L34). Could we possibly reuse this one instead, cause it looks like you are doing very similar thing here.

Reply:
`transformRule` is fine as well. we have to do this because we are fetching rules directly from the API and alerting API have some inconsistency in terms of what they expose on client/server. so we need to use this utility.